src/uid2ApiClient.ts

```diff
@@ -237,25 +251,41 @@ export class Uid2ApiClient {
       if (req.readyState !== req.DONE) return;
       this._requestsInFlight = this._requestsInFlight.filter((r) => r !== req);
       try {
-        if (req.status !== 200) {
-          const response = JSON.parse(req.responseText) as unknown;
-          const result = this.ResponseToCstgResult(response);
-          if (typeof result === "string") rejectPromise(result);
-          else resolvePromise(result);
-        } else {
+        if (req.status === 200) {
           const encodedResp = base64ToBytes(req.responseText);
           const decrypted = await box.decrypt(
             encodedResp.slice(0, 12),
             encodedResp.slice(12)
           );
           const decryptedResponse = new TextDecoder().decode(decrypted);
           const response = JSON.parse(decryptedResponse) as unknown;
-          const result = this.ResponseToCstgResult(response);
-          if (typeof result === "string") rejectPromise(result);
-          else resolvePromise(result);
+          if (isCstgApiSuccessResponse(response)) {
+            resolvePromise({
+              status: "success",
+              identity: response.body,
+            });
+          } else {
+            // A 200 should always be a success response.
+            // Something has gone wrong.
+            rejectPromise(
+              `API error: Response body was invalid for HTTP status 200: ${decryptedResponse}`
+            );
+          }
+        } else if (req.status === 400) {
```

Comment:
i return http 403 for domain name check error [here](https://github.com/IABTechLab/uid2-operator/pull/166/files#diff-4253d50195965a8f4ceaedaf401049047f4b84a1158cf9575d805153da945b4fR347)

should i change it to 400 or should you add 403 here as well?

Reply:
I'm happy to add a 403 check here.

Reply:
@mcollins-ttd ok please do